app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    list_of_args = {}
+
+    def inner(*args, **kwargs) -> Callable:
```

Comment:
You could use a decorator from `functools` for `inner` function to preserve the naming and docstring of the decorated function.